visualizer/flatten.js

```diff
@@ -96,6 +96,27 @@ function Node(node) {
   // Convert before and after time
   this.before = node.before.map((v) => v * ns2s);
   this.after = node.after.map((v) => v * ns2s);
+  this.unref = node.unref.map((v) => v * ns2s);
+  this.ref = node.ref.map((v) => v * ns2s);
+
+  // Compile a list of state changes
+  this.stateChanges = []
+  for (const v of this.before) {
+    this.stateChanges.push([v, 'before'])
```

Comment:
Could we use objects, so I don't have to remember if the timestamp is the first or second index?